@claude Test failed with the following log. `anytype` in return type is only supported in inline function:
src/query/filter.zig:923:62: error: expected return type expression, found 'anytype'
    comptime getStorageFn: fn (@TypeOf(self), comptime type) anytype,
                                                             ^~~~~~~

--- a/src/query/filter.zig
+++ b/src/query/filter.zig
@@ -920,7 +920,7 @@ inline fn filterWithModifiers(
     comptime fields: []const StructField,
     entity: Entity,
     self: anytype,
-    comptime getStorageFn: fn (@TypeOf(self), comptime type) anytype,
+    comptime getStorageFn: anytype,
 ) bool {
     return inline for (fields) |field| {
         const T, const modifier_type = extractType(field.type);

--- a/src/query/filter.zig
+++ b/src/query/filter.zig
@@ -916,7 +916,7 @@ fn extractFree(comptime T: type) type {
 ///   - entity: The entity to check
 ///   - self: The query instance (Query or TagQuery)
 ///   - getStorageFn: A comptime function that returns storage pointer for a given type
-fn filterWithModifiers(
+inline fn filterWithModifiers(
     comptime fields: []const StructField,
     entity: Entity,
     self: anytype,

